internal/services/access_flow.go

```diff
@@ -0,0 +1,476 @@
+	if len(diagnostics) > 0 {
+		return nil, diagnostics
+	}
+
+	trigger := apono.UpdateAccessFlowV1Trigger{
+		Type:      updateAccessFlowRequest.Trigger.Type,
+		Timeframe: updateAccessFlowRequest.Trigger.Timeframe,
+	}
+
+	var approvers []apono.ApproverV1
+	if updateAccessFlowRequest.Approvers == nil {
+		approvers = make([]apono.ApproverV1, 0)
+	} else {
+		approvers = updateAccessFlowRequest.Approvers
+	}
+
+	data := apono.UpdateAccessFlowV1{
+		Name:               *apono.NewNullableString(&updateAccessFlowRequest.Name),
+		Active:             *apono.NewNullableBool(&updateAccessFlowRequest.Active),
+		RevokeAfterInSec:   *apono.NewNullableInt32(&updateAccessFlowRequest.RevokeAfterInSec),
+		Trigger:            *apono.NewNullableUpdateAccessFlowV1Trigger(&trigger),
+		Grantees:           updateAccessFlowRequest.Grantees,
+		Approvers:          approvers,
+		IntegrationTargets: updateAccessFlowRequest.IntegrationTargets,
+		Settings:           updateAccessFlowRequest.Settings,
+	}
+
+	return &data, nil
+
+}
+
+func convertToIntegrationTargetModel(ctx context.Context, integrationTarget *apono.AccessTargetIntegrationV1, availableIntegrations []apono.Integration) (*models.IntegrationTarget, diag.Diagnostics) {
+	for _, integration := range availableIntegrations {
+		if integration.Id == integrationTarget.GetIntegrationId() {
+			resourceIncludeFilters := convertTagsToFiltersModel(integrationTarget.GetResourceTagIncludes())
+			resourceExcludeFilters := convertTagsToFiltersModel(integrationTarget.GetResourceTagExcludes())
+
+			permissions, diagnostics := types.SetValueFrom(ctx, types.StringType, integrationTarget.GetPermissions())
+			if len(diagnostics) > 0 {
+				return nil, diagnostics
+			}
+
+			return &models.IntegrationTarget{
+				Name:                   types.StringValue(integration.GetName()),
+				ResourceType:           types.StringValue(integrationTarget.GetResourceType()),
+				ResourceIncludeFilters: resourceIncludeFilters,
+				ResourceExcludeFilters: resourceExcludeFilters,
+				Permissions:            permissions,
+			}, nil
+		}
+	}
+
+	diagnostics := diag.Diagnostics{}
+	diagnostics.AddError("Client Error", fmt.Sprintf("Failed to get integration: %s", integrationTarget.GetIntegrationId()))
+	return nil, diagnostics
+
+}
+
+func convertToIdentityModel(ctx context.Context, identityId string, identityType string, aponoClient *apono.APIClient, availableIdentities []apono.IdentityModel2) (*models.Identity, diag.Diagnostics) {
+	switch identityType {
+	case "user":
+		user, _, err := aponoClient.UsersApi.GetUser(ctx, identityId).Execute()
+		if err != nil {
+			return nil, utils.GetDiagnosticsForApiError(err, "list", "user", identityId)
+		}
+		return &models.Identity{
+			Name: types.StringValue(user.GetEmail()),
+			Type: types.StringValue("user"),
+		}, nil
+
+	case "group", "context_attribute":
+		for _, identity := range availableIdentities {
```

Comment:
why for users you fetch them 1 by 1 and for other identities you get from the already resolved map?
it should be same endpoint and same data, and even if not, you want to prefetch both lists once,
making a get per item per access flow is not an option, this means the terraform work times at scale will be horrible